File: lab2/Evaluations.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Data.Entity.Spatial

Partial Public Class Evaluations
    <Key>
    <DatabaseGenerated(DatabaseGeneratedOption.None)>
    Public Property ID_Student As Integer

    <Column(TypeName:="date")>
    Public Property Date_Exam1 As Date?

    Public Property ID_Subjects1 As Integer?

    Public Property Evaluations1 As Integer?

    <Column(TypeName:="date")>
    Public Property Date_Exam2 As Date?

    Public Property ID_Subjects2 As Integer?

    Public Property Evaluations2 As Integer?

    Public Property Average_Score As Double?

    Public Overridable Property Students As Students

    Public Overridable Property Subjects As Subjects

    Public Overridable Property Subjects1 As Subjects
End Class

File: Labs2/Models/Evaluations.cs
namespace Labs2.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    public partial class Evaluations
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int ID_Student { get; set; }

        [Column(TypeName = "date")]
        public DateTime? Date_Exam1 { get; set; }

        public int? ID_Subjects1 { get; set; }

        public int? Evaluations1 { get; set; }

        [Column(TypeName = "date")]
        public DateTime? Date_Exam2 { get; set; }

        public int? ID_Subjects2 { get; set; }

        public int? Evaluations2 { get; set; }

        public double? Average_Score { get; set; }

        public virtual Students Students { get; set; }

        public virtual Subjects Subjects { get; set; }

        public virtual Subjects Subjects1 { get; set; }
    }
}
